### PR TITLE
EVG-5369 create data models for triggers

### PR DESCRIPTION
**Pull request by john-m-liu**
most of this is actually "handle new requester types in existing queries"

### CODE REVIEW

File: globals.go [hunk 1]
```diff
@@ -216,6 +220,8 @@ const (
 	PatchVersionRequester       = "patch_request"
 	GithubPRRequester           = "github_pull_request"
 	RepotrackerVersionRequester = "gitter_request"
+	TriggerRequester            = "trigger_request"
+	UserInitiatedRequester      = "user_initiated"
```
Comment (tychoish) [inline globals.go]: I don't like that they all end with _request, but... 

Feels weird that user_initiated is different. 

Also user initiated feels like the wrong descriptor for "projectless/freeagent" also I suspect that user-initiated will be one of the key ways that users add new repository types and do cool workflow things, and it'll feel as awkward as "gitter", so I want to try and see if we can sand off some of the rough corners.

File: model/event/subscriptions.go [hunk 2]
```diff
@@ -456,6 +456,7 @@ func NewPatchOutcomeSubscriptionByOwner(owner string, sub Subscriber) Subscripti
 	return NewSubscriptionByOwner(owner, sub, ResourceTypePatch, triggerOutcome)
 }
 
+// TODO PM-794: also create one of these for requester = trigger?
```
Comment (tychoish) [inline model/event/subscriptions.go]: PM tickets aren't public, just create the ticket and avoid the comment.

File: rest/model/build.go [hunk 3]
```diff
@@ -12,8 +12,9 @@ import (
 )
 
 var (
-	commitOrigin = "commit"
-	patchOrigin  = "patch"
+	commitOrigin  = "commit"
+	patchOrigin   = "patch"
+	triggerOrigin = "trigger"
```
Comment (tychoish) [inline rest/model/build.go]: what about api'd ones?

File: globals.go (hunk 1 above)
Comment (john-m-liu) [reply globals.go]: I renamed it to ad hoc, though if you can think of a description that encompasses all the relevant use cases we should change it to that before it starts being used

File: model/event/subscriptions.go (hunk 2 above)
Comment (john-m-liu) [reply model/event/subscriptions.go]: oops I had already opened EVG-5399 but forgot to remove this

File: rest/model/build.go (hunk 3 above)
Comment (john-m-liu) [reply rest/model/build.go]: added

**Comment by tychoish** [comment]
fix lint?

**Pull request merged**